parts/k8s/cloud-init/artifacts/cse_install.sh

```diff
@@ -385,9 +385,9 @@ pullContainerImage() {
 }
 
 cleanUpContainerImages() {
-    docker rmi $(docker images --format '{{OpenBraces}}.Repository{{CloseBraces}}:{{OpenBraces}}.Tag{{CloseBraces}}' | grep -v "${KUBERNETES_VERSION}$" | grep 'hyperkube') &
-    docker rmi $(docker images --format '{{OpenBraces}}.Repository{{CloseBraces}}:{{OpenBraces}}.Tag{{CloseBraces}}' | grep -v "${KUBERNETES_VERSION}$" | grep 'cloud-controller-manager') &
-    docker rmi $(docker images --format '{{OpenBraces}}.Repository{{CloseBraces}}:{{OpenBraces}}.Tag{{CloseBraces}}' | grep -v "${ETCD_VERSION}$" | grep 'etcd') &
+    docker rmi $(docker images --format '{{OpenBraces}}.Repository{{CloseBraces}}:{{OpenBraces}}.Tag{{CloseBraces}}' | grep -v "${KUBERNETES_VERSION}" | grep 'hyperkube') &
```

Comment:
With this change, could `1.17.1 be confused with `1.17.10`?

Reply:
Yes

Reply:
I can add a suffix like "-" or "_" to avoid that. But leaving one extra image on the disk shouldn't be a big problem right?

Reply:
> But leaving one extra image on the disk shouldn't be a big problem right 
We should fix this, any unexpected behavior can cause bugs down the road even if the impact is small right now

> I can add a suffix like "-" or "_" to avoid that
Wouldn't that delete all the regular images (not AKS patched) images?

Reply:
what we need is a behavior that does: 
Delete all images unless
- it ends with {VERSION}
OR
- it contains {VERSION}-
does that make sense?

Reply:
yes, I would also add {VERSION}_ since we have that case too.

Reply:
tony@test0308vhd:~$ sudo docker images --format '{{.Repository}}:{{.Tag}}' | grep 13 | grep hyperkube
mcr.microsoft.com/oss/kubernetes/hyperkube:v1.13.12_f0.0.2
mcr.microsoft.com/oss/kubernetes/hyperkube:v1.13.11_v0.0.5
mcr.microsoft.com/oss/kubernetes/hyperkube:v1.13.1
mcr.microsoft.com/oss/kubernetes/hyperkube:v1.13.1-f
mcr.microsoft.com/oss/kubernetes/hyperkube:v1.13.10_v0.0.5
mcr.microsoft.com/oss/kubernetes/hyperkube:v1.13.1_v
tony@test0308vhd:~$  sudo docker images --format '{{.Repository}}:{{.Tag}}' | grep  -vE "1.13.1$|1.13.1-|1.13.1_" | grep hyperkube | grep 13
mcr.microsoft.com/oss/kubernetes/hyperkube:v1.13.12_f0.0.2
mcr.microsoft.com/oss/kubernetes/hyperkube:v1.13.11_v0.0.5
mcr.microsoft.com/oss/kubernetes/hyperkube:v1.13.10_v0.0.5